lua/DCS-gRPC/methods/controllers.lua

```diff
@@ -30,3 +30,63 @@ GRPC.methods.setAlarmState = function(params)
 
   return GRPC.success({})
 end
+
+GRPC.methods.getDetectedTargets = function(params)
+  local unit = Unit.getByName(params.unitName)
+  if unit == nil then
+    return GRPC.errorNotFound("Could not find unit with name '" .. params.unitName .. "'")
+  end
+
+  local controller = Unit.getController(unit)
+  local targets = controller:getDetectedTargets()
+
+  if targets == nil then
+    return GRPC.success({
+      contacts = targets
+    })
+  end
+
+  local results = {}
+
+  for i, contact in ipairs(targets) do
+    local category = Object.getCategory(contact.object)
+
+    if category == nil then
+      return GRPC.errorNotFound("Could not find target with id '" .. contact.object:getID() .. "'")
+    end
+
+    local result = {
+      distance = contact.distance,
+      id = contact.object.id_,
+      visible = contact.visible,
+      target = {}
+    }
+
+    --If target is a unit
+    if category == 1 then
```

Comment:
Another option would be to use https://github.com/DCS-gRPC/rust-server/blob/main/lua/DCS-gRPC/exporters/object.lua#L111 

If details not needed. Export using the "Unknown" exporter type.

Otherwise use this if statement.

Reply:
Odd problem with using the `Unknown` exporter. It's returning a type of `0`:

```    
    {
      "id": 16778240,
      "visible": true,
      "distance": true,
      "object": {
        "name": "Aerial-1-1",
        "type": 0
      }
```

But neither the exporter method nor the Unknown common proto has a type field. This is not causing an error.

```
message Unknown {
  string name = 1;
}
```

```lua
GRPC.exporters.unknown = function(object)
  return {
    name = object:getName(),
  }
end
```

So I can leave this in but it's technically inaccurate or misleading.

Reply:
i don't follow with innacurate since there is no type in the RPC response which I think is fine?